File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/Params.java
package com.algolia.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

/** Additional search parameters. Any valid search parameter is allowed. */
public class Params {

  @SerializedName("query")
  private String query;

  @SerializedName("automaticFacetFilters")
  private List<AutomaticFacetFilter> automaticFacetFilters = null;

  @SerializedName("automaticOptionalFacetFilters")
  private List<AutomaticFacetFilter> automaticOptionalFacetFilters = null;

  public Params query(String query) {
    this.query = query;
    return this;
  }

  /**
   * Query string.
   *
   * @return query
   */
  @javax.annotation.Nullable
  public String getQuery() {
    return query;
  }

  public void setQuery(String query) {
    this.query = query;
  }

  public Params automaticFacetFilters(
    List<AutomaticFacetFilter> automaticFacetFilters
  ) {
    this.automaticFacetFilters = automaticFacetFilters;
    return this;
  }

  public Params addAutomaticFacetFiltersItem(
    AutomaticFacetFilter automaticFacetFiltersItem
  ) {
    if (this.automaticFacetFilters == null) {
      this.automaticFacetFilters = new ArrayList<>();
    }
    this.automaticFacetFilters.add(automaticFacetFiltersItem);
    return this;
  }

  /**
   * Names of facets to which automatic filtering must be applied; they must match the facet name of
   * a facet value placeholder in the query pattern.
   *
   * @return automaticFacetFilters
   */
  @javax.annotation.Nullable
  public List<AutomaticFacetFilter> getAutomaticFacetFilters() {
    return automaticFacetFilters;
  }

  public void setAutomaticFacetFilters(
    List<AutomaticFacetFilter> automaticFacetFilters
  ) {
    this.automaticFacetFilters = automaticFacetFilters;
  }

  public Params automaticOptionalFacetFilters(
    List<AutomaticFacetFilter> automaticOptionalFacetFilters
  ) {
    this.automaticOptionalFacetFilters = automaticOptionalFacetFilters;
    return this;
  }

  public Params addAutomaticOptionalFacetFiltersItem(
    AutomaticFacetFilter automaticOptionalFacetFiltersItem
  ) {
    if (this.automaticOptionalFacetFilters == null) {
      this.automaticOptionalFacetFilters = new ArrayList<>();
    }
    this.automaticOptionalFacetFilters.add(automaticOptionalFacetFiltersItem);
    return this;
  }

  /**
   * Same syntax as automaticFacetFilters, but the engine treats the filters as optional.
   *
   * @return automaticOptionalFacetFilters
   */
  @javax.annotation.Nullable
  public List<AutomaticFacetFilter> getAutomaticOptionalFacetFilters() {
    return automaticOptionalFacetFilters;
  }

  public void setAutomaticOptionalFacetFilters(
    List<AutomaticFacetFilter> automaticOptionalFacetFilters
  ) {
    this.automaticOptionalFacetFilters = automaticOptionalFacetFilters;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Params params = (Params) o;
    return (
      Objects.equals(this.query, params.query) &&
      Objects.equals(
        this.automaticFacetFilters,
        params.automaticFacetFilters
      ) &&
      Objects.equals(
        this.automaticOptionalFacetFilters,
        params.automaticOptionalFacetFilters
      )
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(
      query,
      automaticFacetFilters,
      automaticOptionalFacetFilters
    );
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Params {\n");
    sb.append("    query: ").append(toIndentedString(query)).append("\n");
    sb
      .append("    automaticFacetFilters: ")
      .append(toIndentedString(automaticFacetFilters))
      .append("\n");
    sb
      .append("    automaticOptionalFacetFilters: ")
      .append(toIndentedString(automaticOptionalFacetFilters))
      .append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/Params.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * Params Class Doc Comment
 *
 * @category Class
 * @description Additional search parameters. Any valid search parameter is allowed.
 *
 * @package Algolia\AlgoliaSearch
 */
class Params extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'query' => 'string',
        'automaticFacetFilters' => '\Algolia\AlgoliaSearch\Model\Search\AutomaticFacetFilter[]',
        'automaticOptionalFacetFilters' => '\Algolia\AlgoliaSearch\Model\Search\AutomaticFacetFilter[]',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'query' => null,
        'automaticFacetFilters' => null,
        'automaticOptionalFacetFilters' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'query' => 'setQuery',
        'automaticFacetFilters' => 'setAutomaticFacetFilters',
        'automaticOptionalFacetFilters' => 'setAutomaticOptionalFacetFilters',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'query' => 'getQuery',
        'automaticFacetFilters' => 'getAutomaticFacetFilters',
        'automaticOptionalFacetFilters' => 'getAutomaticOptionalFacetFilters',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['query'])) {
            $this->container['query'] = $data['query'];
        }
        if (isset($data['automaticFacetFilters'])) {
            $this->container['automaticFacetFilters'] = $data['automaticFacetFilters'];
        }
        if (isset($data['automaticOptionalFacetFilters'])) {
            $this->container['automaticOptionalFacetFilters'] = $data['automaticOptionalFacetFilters'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets query
     *
     * @return string|null
     */
    public function getQuery()
    {
        return $this->container['query'] ?? null;
    }

    /**
     * Sets query
     *
     * @param string|null $query query string
     *
     * @return self
     */
    public function setQuery($query)
    {
        $this->container['query'] = $query;

        return $this;
    }

    /**
     * Gets automaticFacetFilters
     *
     * @return \Algolia\AlgoliaSearch\Model\Search\AutomaticFacetFilter[]|null
     */
    public function getAutomaticFacetFilters()
    {
        return $this->container['automaticFacetFilters'] ?? null;
    }

    /**
     * Sets automaticFacetFilters
     *
     * @param \Algolia\AlgoliaSearch\Model\Search\AutomaticFacetFilter[]|null $automaticFacetFilters names of facets to which automatic filtering must be applied; they must match the facet name of a facet value placeholder in the query pattern
     *
     * @return self
     */
    public function setAutomaticFacetFilters($automaticFacetFilters)
    {
        $this->container['automaticFacetFilters'] = $automaticFacetFilters;

        return $this;
    }

    /**
     * Gets automaticOptionalFacetFilters
     *
     * @return \Algolia\AlgoliaSearch\Model\Search\AutomaticFacetFilter[]|null
     */
    public function getAutomaticOptionalFacetFilters()
    {
        return $this->container['automaticOptionalFacetFilters'] ?? null;
    }

    /**
     * Sets automaticOptionalFacetFilters
     *
     * @param \Algolia\AlgoliaSearch\Model\Search\AutomaticFacetFilter[]|null $automaticOptionalFacetFilters same syntax as automaticFacetFilters, but the engine treats the filters as optional
     *
     * @return self
     */
    public function setAutomaticOptionalFacetFilters($automaticOptionalFacetFilters)
    {
        $this->container['automaticOptionalFacetFilters'] = $automaticOptionalFacetFilters;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

